pyat/at/physics/frequency_maps.py

```diff
@@ -0,0 +1,283 @@
+"""
+Frequency analysis (FMAP) using .harmonic_analysis lib
+and pyat parallel tracking (patpass)
+"""
+
+# orblancog
+# generates the frequency and diffusion map for a given ring
+# 2023jan16 tracking is parallel (patpass), frequency analysis is serial
+# 2022jun07 serial version
+
+from at.tracking import patpass
+from at.physics import find_orbit
+import numpy
+
+# Jaime Coello de Portugal (JCdP) frequency analysis implementation
+from .harmonic_analysis import get_tunes_harmonic
+
+__all__ = ['fmap_parallel_track']
+
+
+def fmap_parallel_track(ring,
+                        coords=[-10, 10, -10, 10],
+                        steps=[100, 100],
+                        scale='linear',
+                        turns=512,
+                        ncpu=30,
+                        co=True,
+                        add_offset6D=numpy.zeros((6, 1)),
```

Comment:
Why not
```python
add_offset6D=numpy.zeros(6),
```
Later on, you handle it as a 1D array

Reply:
Implemented in commit https://github.com/atcollab/at/pull/556/commits/6503cf4a5baefdf74176a3feb77b4c5921160185